src/chainparamsbase.cpp

```diff
@@ -17,7 +17,8 @@ void SetupChainParamsBaseOptions(ArgsManager& argsman)
     argsman.AddArg("-regtest", "Enter regression test mode, which uses a special chain in which blocks can be solved instantly. "
                  "This is intended for regression testing tools and app development. Equivalent to -chain=regtest.", ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::CHAINPARAMS);
     argsman.AddArg("-testactivationheight=name@height.", "Set the activation height of 'name' (segwit, bip34, dersig, cltv, csv). (regtest-only)", ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::DEBUG_TEST);
-    argsman.AddArg("-testnet", "Use the test chain. Equivalent to -chain=test.", ArgsManager::ALLOW_ANY, OptionsCategory::CHAINPARAMS);
+    argsman.AddArg("-testnet", "Use the testnet3 chain. Equivalent to -chain=test. Support for testnet3 is deprecated and will be removed with the next release. Consider moving to testnet4 now by using -testnet4.", ArgsManager::ALLOW_ANY, OptionsCategory::CHAINPARAMS);
```

Comment:
Are we sure that we will drop support next release? Otherwise it might be better to be a bit less specific:

```suggestion
    argsman.AddArg("-testnet", "Use the testnet3 chain. Equivalent to -chain=test. Support for testnet3 is deprecated and will be removed in a future release. Consider moving to testnet4 now by using -testnet4.", ArgsManager::ALLOW_ANY, OptionsCategory::CHAINPARAMS);
```

Reply:
I think it's better to be precise and we could still shift it if that becomes absolutely necessary for a good reason. It seems better to have people prepare for it to be dropped in the next release unless there is good reason brought forward for shifting. This is better than the alternative that we are vague and then people claim last minute we can not do it in the next release already because we were not clear enough. I don't want to give an invite to the "Testnet3 is the real testnet" crowd :)

Reply:
I was thinking of business to business companies using Testnet3 as a development environment. They may have dozens of enterprise clients that have built Testing infrastructure on Testnet3 and serve transaction data by interfacing with Bitcoin Core. Even if the service provider is willing to upgrade, it can be a pain to get the customers to update (speaking from experience). I’m fine with deprecating Testnet3, but I would suggest that the option should only be removed from Bitcoin Core with v29.

I also noticed that the two mentions of future deprecation are not completely consistent, but that could be addressed in a follow-up or isn’t that big a deal anyhow.

src/chainparamsbase.cpp:
```c++
   argsman.AddArg("-testnet", "Use the testnet3 chain. Equivalent to -chain=test. Support for testnet3 is deprecated and will be removed with the next release. Consider moving to testnet4 now by using -testnet4.", ArgsManager::ALLOW_ANY, OptionsCategory::CHAINPARAMS);
```

src/init.cpp:
```c++
    // Testnet3 deprecation warning
    if (chain == ChainType::TESTNET) {
        LogInfo("Warning: Support for testnet3 is deprecated and will be removed in an upcoming release. Consider switching to testnet4.\n");
    }
```

Reply:
Ok, unless I have to re-touch here I will make it consistent with the follow-up when I add the release notes. 

> I’m fine with deprecating Testnet3, but I would suggest that the option should only be removed from Bitcoin Core with v29.

That's my intention, since we are adding Testnet4 for v28 the next version I am referring to is v29. Or were you concerned with removing it from master right after v28 is tagged? I would be fine to aim for a removal with v30 instead giving ~1 year of deprecation period. But I really think it serves everyone best if we make are clear on the intended timeline right away to manage expectations and invite people to speak up as early as possible if they have an issue with it. I prefer a faster timeline but indeed I don't have a good overview of such b2b Testnet3 infrastructure and how much work it is to move it.

Reply:
@sipa Timeline sounds good to me, but why not be precise in the warning and say something like "deprecated in the next version and removed in the version after"?

Reply:
> Ok, unless I have to re-touch here I will make it consistent in the follow-up where I add the release notes.
> 
> > I’m fine with deprecating Testnet3, but I would suggest that the option should only be removed from Bitcoin Core with v29.
> 
> That's my intention, since we are adding Testnet4 for v28 the next version I am referring to is v29. Or were you concerned with removing it from master right after v28 is tagged? I would be fine to aim for a removal with v30 instead giving ~1 year of deprecation period. But I really think it serves everyone best if we make are clear on the intended timeline right away to manage expectations and invite people to speak up as early as possible if they have an issue with it. I prefer a faster timeline but indeed I don't have a good overview of such b2b Testnet3 infrastructure and how much work it is to move it.

Err, I meant not in 29, but in 30. :facepalm:

Reply:
If it's being considered for testnet3, I'd recommend a fairly long deprecation cycle, at least 2 releases.

Reply:
why not just keep it ? why act like communist and remove things you desire ~ 

alot of data and infrastructure already on testnet3, why force people your own desires ~

Reply:
> @fjahr I guess because I feel it's hard to make concrete promises about something that far out. If we _know_ we will remove testnet3 in v30, we should just deprecate it in v28 already (including a `-deprecatedrpc` requirement), rather than saying it will be deprecated. If not, the best we can do is announce an intent that it will be removed in the future, without concrete timing.

what you mean testnet3 gone ? you seriously wants to delete it~ you can use testnet3 as reference for old scripts thats been used ... 

stop actting like you make decisions ~

Reply:
> what you mean testnet3 gone ? you seriously wants to delete it~ you can use testnet3 as reference for old scripts thats been used ...

No worries, you can continue to use Testnet 3 as long as you desire. It’s just not going to be supported by Bitcoin Core starting with some upcoming version.

Reply:
> > @fjahr I guess because I feel it's hard to make concrete promises about something that far out. If we _know_ we will remove testnet3 in v30, we should just deprecate it in v28 already (including a `-deprecatedrpc` requirement), rather than saying it will be deprecated. If not, the best we can do is announce an intent that it will be removed in the future, without concrete timing.
> 
> what you mean testnet3 gone ? you seriously wants to delete it~ you can use testnet3 as reference for old scripts thats been used ...
> 
> stop actting like you make decisions ~

> I don't want to give an invite to the "Testnet3 is the real testnet" crowd :)

Q.E.D.

Reply:
@boring877 Your comments are off topic and inflammatory, resulting in a 24 ban from the organization to cool off. I'm not going to delete your comments since this PR is closed anyway, and contributors offered informational replies. To appeal a moderation decision, open an issue in bitcoin-core/meta

Reply:
Testnet3 is always available from the right branch and can be forked by any interested community.

Consensus is that Testnet4 will be better for Bitcoin testing.

There is still value in Testnet3, especially for observing Bitcoin beyond 2.8 million blocks, understanding reorgs, and exploring reorg protection strategies.

If there's a community for Testnet3, fork it, create a website, and get folks to run nodes.

Bitcoin devs have decided that Testnet3 is getting harder to use, and Testnet4 should be easier. If you want to keep the Testnet3 experiment going, please do. I think it's worthwhile, but the approach above is a reasonable path, IMHO.